### ISSUE TITLE
🚀  Reduce memory footprint

### ISSUE THREAD

**Issue by ReneWerner87** [gofiber/fiber#1799]
this issue is about collecting and implementing ideas for reducing the initial memory footprint.

Example app:
```go
package main

import (
    "log"

    "github.com/gofiber/fiber/v2"
    "github.com/gofiber/fiber/v2/middleware/monitor"
)


func main() {
    app := fiber.New()

    app.Get("/dashboard", monitor.New())

    log.Fatal(app.Listen(":3000"))
}
```

Monitor:
![image](https://user-images.githubusercontent.com/7063188/156313890-3862e990-167b-4676-8d4c-706b00519954.png)
12.5 MB
---
Example app 2:
```go
package main

import (
    "log"

    "github.com/gofiber/fiber/v2"
)


func main() {
    app := fiber.New()

    app.Get("/test", func(ctx *fiber.Ctx) error {
        return ctx.SendString("OK")
    })

    log.Fatal(app.Listen(":3000"))
}
```

![image](https://user-images.githubusercontent.com/7063188/156314792-032350a8-dfc5-47b5-a4bb-85ffd68c86f5.png)

Activity monitor:
![image](https://user-images.githubusercontent.com/7063188/156314673-3b6adb03-fa75-42bd-b080-74afa7c1007b.png)
9.3 MB
---
Fasthttp:
```go
package main

import (
    "log"

    "github.com/valyala/fasthttp"
)


func main() {

    requestHandler := func(ctx *fasthttp.RequestCtx) {
        ctx.SetBodyString("OK")
    }

    log.Fatal(fasthttp.ListenAndServe(":3000", requestHandler))
}
```

Activity monitor:
![image](https://user-images.githubusercontent.com/7063188/156315599-b7d3b0ec-6f2c-47bb-b944-38c4b4991f91.png)
3.3 MB
---
Would be good if we could reduce the initial memory consumption.

Any findings or ideas ?

**Comment by ReneWerner87** [comment]
![profile001](https://user-images.githubusercontent.com/7063188/156322115-02c06ad8-ffc4-4ae1-a83e-bde41b2e1e7b.png)

**Comment by dranikpg** [comment]
I sadly doubt there is much left to imoprove in fiber. I've runned all examples on latest fiber on ubuntu and all large allocations are made either by the runtime or fasthttp. However, there are some interesting details about fasthttp.

Fasthttp tries to keep zero allocations on hot paths, so it does some things ahead. For example, wrapped functions for parallel usage are created for its gzip/deflate/brotili writers via [stackless.NewFunc](https://github.com/valyala/fasthttp/blob/2a6f7db5bbc4d7c11f1ccc0cb827e145b9b7d7ea/stackless/func.go#L21). This function creates a work channel with a large capacity `make(chan *funcWork, runtime.GOMAXPROCS(-1)*2048)`, which is about 8 (pointer) * 16 (my cores) * 2048 = 0.25mb. All this a few times for different kinds of encoders amounts up to over 1 mb

![Screenshot_20220502_195308](https://user-images.githubusercontent.com/20553775/166296553-e47e7f7a-382f-47c5-ad87-fb8bc3c055b3.png)

Stripping out sync.Pools from fasthttp's server reduced it even further (thats why acquireCtx is on the list)

So in the end I got 

![Screenshot_20220502_204054](https://user-images.githubusercontent.com/20553775/166297587-764e529d-268f-4c48-b4de-2c602dda022b.png)
